### PR TITLE
Eradicate old-style cvars (vmCvar_t)

### PR DESCRIPTION
**Pull request by slipher**

### CODE REVIEW

File: src/sgame/sg_client.cpp [hunk 1]
```diff
@@ -1126,8 +1126,7 @@ const char *ClientConnect( int clientNum, bool firstTime )
 	// check for a password
 	value = Info_ValueForKey( userinfo, "password" );
 
-	if ( g_password.string[ 0 ] && Q_stricmp( g_password.string, "none" ) &&
-	     strcmp( g_password.string, value ) != 0 )
+	if ( g_needpass.Get() && !Str::IsIEqual( g_password.Get(), value ) )
```
Comment (necessarily-equal) [inline src/sgame/sg_client.cpp:1129]: Do we really want to check passwords in a case-insensitive way?

```suggestion
	if ( g_needpass.Get() && g_password.Get() != value )
```

File: src/cgame/cg_main.cpp [hunk 2]
```diff
@@ -417,21 +372,9 @@ CG_UpdateCvars
...
-		{
-			trap_Cvar_Update( cv->vmCvar );
-		}
-	}
-
-	// check for modifications here
 	CG_SetPVars();
 	CG_UpdateBuildableRangeMarkerMask();
-}
+ }
```
Comment (necessarily-equal) [inline src/cgame/cg_main.cpp:377]: :eyes:
```suggestion
}
```

**Review by necessarily-equal — APPROVED** [review]
LGTM

File: src/sgame/sg_client.cpp (hunk 1 above)
Comment (slipher) [reply src/sgame/sg_client.cpp]: Nope!

**Review by necessarily-equal — APPROVED** [review]
looked fairly rapidly, LGTM

**Pull request merged**